flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -31,6 +31,10 @@
 	/**
 	 * Asynchronously register a partition and its producer with the shuffle service.
 	 *
+	 * <p>IMPORTANT: Please implement the asynchronous interface in a synchronous way.
+	 * The default scheduler only supports synchronous registration for now.
+	 * Please refer to {@link NettyShuffleMaster#registerPartitionWithProducer} for example.
```

Comment:
I'd prefer to state that "one must return a completed future at the moment". It is clear enough so the user do need to refer to NettyShuffleMaster for the implementation.